plugins/chat/assets/javascripts/discourse/routes/chat-channel.js

```diff
@@ -1,5 +1,26 @@
 import DiscourseRoute from "discourse/routes/discourse";
 import withChatChannel from "./chat-channel-decorator";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
 
 @withChatChannel
-export default class ChatChannelRoute extends DiscourseRoute {}
+export default class ChatChannelRoute extends DiscourseRoute {
+  @service chatThreadsManager;
+  @service chatStateManager;
+
+  @action
+  willTransition(transition) {
+    this.chat.activeThread = null;
+    this.chatStateManager.closeSidePanel();
+
+    if (!transition?.to?.name?.startsWith("chat.")) {
```

Comment:
I had to add this so this navigation spec and some others would pass https://github.com/discourse/discourse/blob/bf2a9063f6af861a0f72f6d7bbf728116c049276/plugins/chat/spec/system/navigation_spec.rb#L264-L264 . It's not needed in my other PR where I covert `activeChannel` to being tracked too. If we merge the other one first I can get rid of it here. https://github.com/discourse/discourse/pull/20237

Reply:
Yes I expect it to have few nice side effects indeed